stix_shifter_modules/msatp/stix_translation/query_constructor.py

```diff
@@ -177,6 +179,8 @@ def _parse_mapped_fields(expression, value, comparator, mapped_fields_array):
                                          ComparisonComparators.Equal, ComparisonComparators.NotEqual]:
                 if is_int_field or is_date_field:
                     mapped_field = 'tostring({mapped_field})'.format(mapped_field=mapped_field)
+                elif expression.object_path == "domain-name:value":
+                    comparator = 'contains'
```

Comment:
little confused about this condition. is `domain-name:value ` a special case. is it only works with `contains` comparator?

Reply:
yes, its a bit confusing. so in msatp RemoteUrl field is used for both url's and domains. In most of the cases its a url, in some a domain. in order for the domain search to work on url's you need to perform a contains operator, on the other hand searching for url requires the equality comparator, so yes - this is a special case for domain-name only

Reply:
okk thanks for the details. i would put a comment for this condition.

Reply:
done